plugins/packages/grpc/lib/index.ts

```diff
@@ -0,0 +1,69 @@
+import { QueryError, QueryResult, QueryService } from '@tooljet-plugins/common';
+import { SourceOptions, QueryOptions } from './types';
+
+import * as grpc from '@grpc/grpc-js';
+import * as protoLoader from '@grpc/proto-loader';
+
+export default class GRPC implements QueryService {
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions, dataSourceId: string): Promise<QueryResult> {
+    const { serviceName, rpc } = queryOptions;
+
+    if (!sourceOptions.url) {
+      throw new QueryError('Missing URL', {}, {});
+    }
+
+    const cwd = process.cwd();
+    const rootDir = cwd.split('/').slice(0, -1).join('/');
+    const protoFilePath = `${rootDir}/protos/service.proto`;
+
+    if (!sourceOptions.protobuf || sourceOptions.protobuf.length === 0) {
```

Comment:
This will always be empty string currently yes? No need to remove, can be placeholder for check later.